rpm/src/main/java/org/eclipse/packager/rpm/deps/Dependencies.java

```diff
@@ -85,7 +85,7 @@ private static void putDependencies ( final Header<RpmTag> header, final Collect
         // first sort
 
         final List<Dependency> deps = new ArrayList<> ( dependencies );
-        Collections.sort ( deps, comparing ( Dependency::getName ).thenComparing ( comparing ( Dependency::getVersion, nullsFirst ( naturalOrder () ) ) ) );
+        Collections.sort ( deps, comparing ( Dependency::getName ).thenComparing ( comparing ( Dependency::getVersion, nullsFirst ( naturalOrder () ) ) ).thenComparing ( comparing ( d -> RpmDependencyFlags.encode ( d.getFlags () ), naturalOrder () ) ) );
```

Comment:
If we made `Dependency` implement `Comparable`, then we could just call `Collections.sort()`.